src/diff.rs

```diff
@@ -214,19 +219,52 @@ fn write_packages_diffln<'a>(
 
   let mut diffs = paths
     .into_iter()
-    .filter_map(|(name, mut versions)| {
+    .map(|(name, mut versions)| {
       deduplicate_versions(&mut versions.old);
       deduplicate_versions(&mut versions.new);
 
       let status = match (versions.old.len(), versions.new.len()) {
         (0, 0) => unreachable!(),
         (0, _) => DiffStatus::Added,
         (_, 0) => DiffStatus::Removed,
-        (..) if versions.old != versions.new => DiffStatus::Changed,
-        (..) => return None,
+        _ => {
+          let mut saw_upgrade = false;
+          let mut saw_downgrade = false;
+
+          for diff in
+            Itertools::zip_longest(versions.old.iter(), versions.new.iter())
+          {
+            match diff {
+              EitherOrBoth::Both(old, new) => {
+                match old.cmp(new) {
+                  cmp::Ordering::Less => saw_upgrade = true,
+                  cmp::Ordering::Greater => saw_downgrade = true,
+                  cmp::Ordering::Equal => {},
+                }
+
+                if saw_upgrade && saw_downgrade {
+                  break;
+                }
+              },
+              EitherOrBoth::Left(_) => {
+                saw_downgrade = true;
+              },
+              EitherOrBoth::Right(_) => {
+                saw_upgrade = true;
+              },
+            }
+          }
+
+          match (saw_upgrade, saw_downgrade) {
+            (true, true) => DiffStatus::Changed,
+            (true, false) => DiffStatus::Upgraded,
+            (false, true) => DiffStatus::Downgraded,
+            _ => unreachable!(),
```

Comment:
```suggestion
            _ => return None,
```

We removed the _ => case before, we can trivially hit this. Need a filter_map